packages/cpp/src/browser/cpp-language-client-contribution.ts

```diff
@@ -28,6 +36,27 @@ export class CppLanguageClientContribution extends BaseLanguageClientContributio
         super(workspace, languages, languageClientFactory);
     }
 
+    protected onReady(languageClient: ILanguageClient): void {
+        super.onReady(languageClient);
+
+        // When the language server is ready, send the active build
+        // configuration so it knows where to find the build commands
+        // (e.g. compile_commands.json).
+        this.onActiveBuildConfigChanged(this.cppBuildConfigurations.getActiveConfig());
+        this.cppBuildConfigurations.onActiveConfigChange(config => this.onActiveBuildConfigChanged(config));
+    }
+
+    async onActiveBuildConfigChanged(config: CppBuildConfiguration | undefined) {
+        const interfaceParams: DidChangeConfigurationParams = {
+            settings: {
+                compilationDatabasePath: config ? config.directory : undefined,
+            },
+        };
+
+        const languageClient = await this.languageClient;
```

Comment:
I wonder if we should have a error logged or UI notification when the directory is defined but does not exist. Maybe it could be Clangd that sends the notification back (window/showMessage?).
Otherwise, it can be a bit confusing to see all the errors in the editor and not realize that the path is just wrong.

Reply:
Good idea.  Is it ok if we do this in a separate patch?

Reply:
yes